### PR TITLE
STR-64 -- BE: Change Roles to Groups to make more sense

### PR DESCRIPTION
**Pull request by ekno27**
# Description
Changed name from roles to groups in any backend file. Also returned the user group to the front end through the ``loginVerification`` route
  
# Testing
``composer dump``
``php artisan migrate:fresh --seed``
check that the databases are ok using the sql database tool of your choice
hit the verifyLoginRoute using email ``nr_cyndee.bellamy@csun.edu`` to see that the group is passed alongside everything else
try to stay awake after eating all that pizza

# Installation
'no'

### CODE REVIEW

**Review by liadGolan — APPROVED** [review]
👍

File: app/Models/UserGroup.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    protected $table = 'user_groups';
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'user_id', 'user_group', 'remember_token', 'display_name',
+    ];
+    public function getUserRole()
```
Comment (frlzjosh) [inline app/Models/UserGroup.php]: change this to `getUserGroup()`

File: database/factories/UserGroupFactory.php [hunk 2]
```diff
@@ -4,10 +4,10 @@
 
 use Faker\Generator as Faker;
 
-$factory->define(App\Models\UserRoles::class, function (Faker $faker) {
+$factory->define(App\Models\UserGroup::class, function (Faker $faker) {
     return [
-        'user_email' => 'someone@face.com',
-        'user_role' => 'comparison',
+        'user_id' => 'someone@face.com',
```
Comment (frlzjosh) [inline database/factories/UserGroupFactory.php]: this should be `members:000___`

File: tests/phpunit/ControllerTests/UserGroupControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        parent::setUp();
+        $this->utility = Mockery::mock(UserGroupContract::class);
+        $this->UserGroupController = new UserGroupController($this->utility);
+    }
+
+    /**
+     * @test
+     */
+    public function getGroup_returns_group_and_email_from_utility_as_array()
+    {
```
Comment (frlzjosh) [inline tests/phpunit/ControllerTests/UserGroupControllerTest.php:29]: you should change the method to `getGroup_returns_group_and_id_from_utility_as_array()`

File: tests/phpunit/ServiceTests/AuthenticationServiceTest.php [hunk 4]
```diff
@@ -33,11 +38,12 @@ public function setUp()
      */
     public function authenticateUser_returns_authenticated_user(){
 //        $data = ['username' => 'steve@csun.edu', 'password' => ''];
-        $returnData = ['user_id'=>'members:100010526',
+        $returnData = ['user_id'=>'members:000022575',
```
Comment (frlzjosh) [inline tests/phpunit/ServiceTests/AuthenticationServiceTest.php:40]: can you take out to the comment if it isn't needed

File: tests/phpunit/ServiceTests/UserGroupServiceTest.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+class UserGroupServiceTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    /**
+     * @test
+     */
+    public function getGroup_returns_group_remember_token_display_name_and_email_from_utility_as_array()
+    {
```
Comment (frlzjosh) [inline tests/phpunit/ServiceTests/UserGroupServiceTest.php:20]: replace "email" with "id" in the method name

File: tests/phpunit/ServiceTests/UserGroupServiceTest.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        ])->save();
+
+        $data = [
+            'user_id' => 'someone@face.com',
+            'user_group' => 'some group',
+            'display_name' => 'Someone Name',
+            'remember_token' => 'toke',
+        ];
+
+        $email = ['user_id' => 'someone@face.com'];
```
Comment (frlzjosh) [inline tests/phpunit/ServiceTests/UserGroupServiceTest.php]: var should be like `$id`

File: tests/phpunit/ServiceTests/UserGroupServiceTest.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        $outputFromUserGroupService = $userGroupService->getGroup($email);
+
```
Comment (frlzjosh) [inline tests/phpunit/ServiceTests/UserGroupServiceTest.php:39]: where you see `user_id` you need to change it to the `members:000__`

**Review by frlzjosh — CHANGES_REQUESTED** [review]
Just a good amount of things that need to be renamed

**Review by frlzjosh — APPROVED** [review]
Nice. We're getting there. It was a tough week, but we persevered

**Review by nercesk — APPROVED** [review]

**Pull request merged**